src/printer/smt2/smt2_printer.cpp

```diff
@@ -624,80 +654,110 @@ void Smt2Printer::toStream(std::ostream& out,
       return;
     }
     break;
-  case kind::APPLY_CONSTRUCTOR: typeChildren = true;
-  case kind::APPLY_TESTER:
-  case kind::APPLY_SELECTOR:
-  case kind::APPLY_SELECTOR_TOTAL:
-  case kind::PARAMETRIC_DATATYPE:
-    break;
 
-  //separation logic
-  case kind::SEP_EMP:
-  case kind::SEP_PTO:
-  case kind::SEP_STAR:
-  case kind::SEP_WAND:out << smtKindString(k) << " "; break;
+    case kind::APPLY_CONSTRUCTOR:
+    {
+      typeChildren = true;
+      const Datatype& dt = Datatype::datatypeOf(n.getOperator().toExpr());
+      if (dt.isTuple())
+      {
+        stillNeedToPrintParams = false;
+        if (dt[0].getNumArgs() == 0)
```

Comment:
The two branches here do the same thing. Simplify this into one statement if this isn't necessary.

Reply:
Do you mean the if-else on lines 661 and 665? The trailing space is the difference - if it is an empty tuple, arguments don't follow mkTuple; otherwise space-separated arguments follow.

Reply:
I see, I would simplify this to print the space in the case:
```
out << "mkTuple" << ( dt[0].getNumArgs()==0 ? "" : " ");
```

Reply:
Neat. Thanks.